internal/initwd/module_install.go

```diff
@@ -49,6 +49,22 @@ type moduleVersion struct {
 	version string
 }
 
+type TypeDiagnosticExtra string
+
+const TypeModuleVersionDeprecation = "module_version_deprecation"
+
+type ModuleVersionDeprecationDiagnosticExtra struct {
+	Type               TypeDiagnosticExtra `json:"type"`
+	Version            string              `json:"version"`
+	SourceName         string              `json:"source_name"`
+	DeprecationMessage string              `json:"deprecation_message"`
+	Link               string              `json:"link"`
+}
+
+func (m ModuleVersionDeprecationDiagnosticExtra) IsPublic() {
+	// NOP
```

Comment:
Not exactly. This is used to only expose the Extra field if it's been given the IsPublic function.

In _internal/tfdiags/diagnostic.go_
```go
// PublicExtraInfo is an interface for marking Extra field that contain public extra information
type PublicExtraInfo interface {
	IsPublic()
}
```

In _internal/command/views/json/diagnostic.go_
```go
        extra := diag.ExtraInfo()
	var publicExtra tfdiags.PublicExtraInfo
	if e, ok := extra.(tfdiags.PublicExtraInfo); ok {
		publicExtra = e
	}

	diagnostic := &Diagnostic{
		Severity: sev,
		Summary:  desc.Summary,
		Detail:   desc.Detail,
		Address:  desc.Address,
		Extra:    publicExtra,
	}
```

Otherwise we'd be needlessly showing these Extra fields when they aren't meant for display.